### PR TITLE
Suggested implementation of open/save file dialogs for Linux using GTK+

### PR DESCRIPTION
**Pull request by SirJonthe**
I implemented open/save file dialogs for Linux using GTK+ 2 (switching to GTK+ 3 should be as simple as making a corresponding change in the build script). Using GTK+ 2 this adds about 2 MB of RAM usage on a cold boot. I'm unsure if you opt to use GTK+ or if you would prefer to make your own custom dialogs, but at least for now you have a working placeholder implementation.

### CODE REVIEW

**Comment by ApoorvaJ** [comment]
In the long run, I'd like to get rid of external linkage and implement the dialogs myself (Blender-style). But I will merge this right now. I agree with you in that it's definitely better than having no dialog boxes at all.

I will test and merge in the coming 2-3 days.

**Comment by ApoorvaJ** [comment]
I tested your version on my Elementary OS setup, and it runs fine, but gives this error on execution:

```
Gtk-Message: Failed to load module "pantheon-filechooser-module"
(papaya:9953): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
(papaya:9953): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
(papaya:9953): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
(papaya:9953): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
(papaya:9953): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
Gtk-Message: Failed to load module "canberra-gtk-module"
```

[Apparently](http://elementaryos.stackexchange.com/questions/1859/gtk-message-failed-to-load-module-pantheon-filechooser-module), Elementary OS likes GTK 3 better. So I installed the necessary libraries and, again it runs fine, but now gives these errors:

```
** (papaya:12499): WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-o7ZOsP34vr: Connection refused
(papaya:12499): Gtk-CRITICAL **: gtk_combo_box_text_insert: assertion 'text != NULL' failed
```

After a bit of research, it seems we are in the domain of nasty Gnome bugs, which we'd like to avoid. This is, after all, a "handmade" project.

In conclusion, I will have a look at this after I am done with my current task - implementing a color picker. In the meantime, if you could confirm that you are not getting any errors on Ubuntu/whatever distro you're using, I will merge the GTK 2 branch.

**Comment by SirJonthe** [comment]
I see no errors in standard output on Ubuntu 14.04. Where are the errors outputted for you?

**Comment by ApoorvaJ** [comment]
I see the errors in the terminal - not during compilation, but during execution. I'm very sure this is a problem with Elementary OS. The program still works, so it's good enough for now.

I also checked the GTK+'s license terms, and according to my limited legal understanding, GTK+'s LGPL will not interfere with Papaya's MIT license, as long as we link dynamically (which we are doing).

So, merging now. As mentioned before, I'll look at this again once I'm done implementing the color picker. Thanks for the contributions! :)
